src/higlass/widget.js

```diff
@@ -1,4 +1,129 @@
 import * as hglib from "https://esm.sh/higlass@1.13?deps=react@17,react-dom@17,pixi.js@6";
+import { v4 } from "https://esm.sh/@lukeed/uuid@2.0.1";
+
+/** @import { HGC, PluginDataFetcherConstructor } from "./types.ts" */
+
+// Make sure plugins are registered and enabled
+window.higlassDataFetchersByType = window.higlassDataFetchersByType ||
+  {};
+
+/**
+ * Create a unique identifier.
+ * @returns {string}
+ */
+function uid() {
+  return v4().split("-")[0];
+}
+
+/**
+ * Make an assertion.
+ *
+ * @param {unknown} expression - The expression to test.
+ * @param {string=} msg - The optional message to display if the assertion fails.
+ * @returns {asserts expression}
+ * @throws an {@link Error} if `expression` is not truthy.
+ */
+function assert(expression, msg = "") {
+  if (!expression) throw new Error(msg);
+}
+
+/**
+ * @template T
+ * @param {import("npm:@anywidget/types").AnyModel} model
+ * @param {unknown} payload
+ * @param {{ timeout?: number }} [options]
+ * @returns {Promise<{ data: T, buffers: DataView[] }>}
+ */
+function send(model, payload, { timeout = 3000 } = {}) {
+  let uuid = uid();
+  return new Promise((resolve, reject) => {
+    let timer = setTimeout(() => {
+      reject(new Error(`Promise timed out after ${timeout} ms`));
+      model.off("msg:custom", handler);
+    }, timeout);
+    /**
+     * @param {{ uuid: string, payload: T }} msg
+     * @param {DataView[]} buffers
+     */
+    function handler(msg, buffers) {
+      if (!(msg.uuid === uuid)) return;
+      clearTimeout(timer);
+      resolve({ data: msg.payload, buffers });
+      model.off("msg:custom", handler);
+    }
+    model.on("msg:custom", handler);
+    model.send({ payload, uuid });
+  });
+}
+
+/**
+ * Transforms the original view config into tracks recognized by the custom data fetcher.
+ *
+ * Finds tracks with `server: 'jupyter'`, removes the key, and adds a `data` object
```

Comment:
Is this still true?

I don't see this in the example notebook.

Reply:
Namely I don't see the track's server being set to Jupyter in https://github.com/higlass/higlass-python/blob/manzt/jupyter-comms/examples/JupyterServer.ipynb

Reply:
The viewconf I see that's produced contains the following data section.

```
            "data": {
              "type": "IPY_MODEL_d3b1fccd786440b8a1b9b47bda3cd168",
              "tilesetUid": "abf80f79ec3e9e899d168f869f624aaf"
            },
```

Reply:
Oh, nvm. Ignore that internal monologue. The track does indeed have `jupyter` as the server:

```
HeatmapTrack(tilesetUid='abf80f79ec3e9e899d168f869f624aaf', server='jupyter', type='heatmap', uid='8c8c58d9', width=None, height=None, options=None, data=None, position=None, transforms=None)
```